Documentation for ACCESS-RAM3 beta release (#857)
ACCESS-NRI/access-hive.org.au: docs/models/run-a-model/run-access-ram.md

[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[model components]: /models/configurations/access-ram/#model-components
[model configurations]: /models/configurations/access-ram
[gadi]: https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview

[![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} {{ ancillary_model }} configuration]({{mosrs_config_ras}}){: class="text-card" style=""}
[![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} {{ regional_model }} configuration]({{mosrs_config_rns}}){: class="text-card"}

# Run ACCESS-RAM3

!!! info
    {{ model }} is a place-holder name until the final release name is chosen.

## About

{{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office (UKMO)](https://www.metoffice.gov.uk/) Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution.<br>

> A description of the model and its components is available in the [ACCESS-RAM3 overview][model configurations].

Author: Edit: "ACCESS-RAM is an ACCESS-NRI supported implementation of the [Unified Model (UM)](https://www.metoffice.gov.uk/research/approach/modelling-systems/unified-model) regional nesting suite for high-resolution regional atmosphere modelling on _Gadi_. A description of the model and its components is available in the [ACCESS-RAM3 overview](https://access-hive.org.au/models/configurations/access-ram/#access-ram3)."

{{ model }} comprises a [Regional Ancillary Suite (RAS)](#ras) (RAS) `{{ ras_id }}` to generate ancillary files needed to run the [Regional Nesting Suite (RNS)](#rns) `{{ rns_id }}` for the domain of interest.

> The instructions below outline how to run {{ model }} using ACCESS-NRI's supported configuration, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

Author: Repetitive mention of Gadi (also mentioned in description above).

> If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

Author: I would delete the above sentence as it is not difficult to find an overview of ACCESS models from the above link to more ACCESS-RAM model detail.

All {{model}} configurations are open source, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1")![CC icon](https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"}![BY icon](https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"} and available on [ACCESS-NRI GitHub]({{github_configs}}).

> {{ model }} release notes are available on the ACCESS-Hive Forum and are updated when new releases are made available.

Author: Perhaps include a link to ACCESS-Hive Forum

    
- **Queue** &rarr; `normalbw`
    
- **Compute Size** &rarr; `tiny` (1 CPU)<br>
    As mentioned above, the ARE VDI session is only needed for setup and startup tasks, which can be easily accomplished with 1 CPU.

- **Project** &rarr; a project of which you are a member.<br>
    The project must have allocated _Service Units (SU)_ to run your simulation. Usually, but not always, this corresponds to your `$PROJECT`.<br>
    For more information, refer to [Join relevant NCI projects](/getting_started/set_up_nci_account#join-relevant-nci-projects).

- **Storage** &rarr; `gdata/access+gdata/hr22+gdata/ki32+gdata/rt52+gdata/zz93+gdata/vk83` (minimum)<br>
    This is a list of all project data storage, joined by plus (`+`) signs, needed for the {{ model }} simulation. In ARE, storage locations need to be explicitly defined to access data from within a VDI instance.<br>
    Every {{ model }} simulation can be unique and input data can originate from various sources. Hence, if your simulation requires data stored in project folders other than the ones listed in the minimum storage above, you need to add those projects to the storage path.<br>

>     For example, if your {{ model }} simulation requires data stored in `/g/data/tm70` and `/scratch/w40`, the following should be added to the minimum storage above: `+gdata/tm70+scratch/w40`

Author: Instead of tm70 and w40 projects, I would use gx60 Atmsophere Working group project in this example, i.e., `/g/data/gx60` and `/scratch/gx60`

```
echo "<name>.${USER}.<project>.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
```

For example, if the user `abc123` started a persistent session named `cylc` under the project `xy00`, the command will be:

<terminal-window data="input">
    <terminal-line>echo "cylc.abc123.xy00.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
    </terminal-line>
    <terminal-line data="input" linedelay="1000">cat ~/.persistent-sessions/cylc-session</terminal-line>
    <terminal-line data="output">cylc.abc123.xy00.ps.gadi.nci.org.au</terminal-line>
</terminal-window>

For more information on how to specify the target session, refer to [Specify Target Session with Cylc7 Suites](https://opus.nci.org.au/display/DAE/Run+Cylc7+Suites#RunCylc7Suites-SpecifyTargetSession).

!!! tip
    You can simultaneously submit multiple {{ model }} runs using the same persistent session without needing to start a new one. Hence, the process of specifying the target persistent session for {{ model }} should only be done once.<br>

>     After specifying the {{ model }} target persistent session the first time, to run {{ model }} you just need to make sure to have an active persistent session named like the specified {{ model }} target persistent session.

Author: Edit: "... should be done only once. Then, to run {{ model }} you just need to check that you have an active persistent session named like the target one you specified above."

#### Terminate a persistent session
!!! tip
    Logging out of a *Gadi* login node or an ARE terminal instance will not affect your persistent session. 

To stop a persistent session, run:
```
persistent-sessions kill <persistent-session-uuid>
```
!!! warning
    When you terminate a persistent session, any model running on that session will stop. Therefore, you should check whether you have any active model runs before terminating a persistent session.

### Rose/Cylc/MOSRS setup

> To run {{ model }}, access to multiple softwares and the Met Office Science Repository Service (MOSRS) is needed.

Author: Edit: plural of software is "software" (i.e. delete the "s")

    <terminal-line>Please enter the MOSRS password for &lt;MOSRS-username&gt;:</terminal-line>
    <terminal-line lineDelay=1500>Checking your credentials using Subversion. Please wait.</terminal-line>
    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: Successfully accessed Subversion with your credentials.</terminal-line>
    <terminal-line lineDelay=100><span style="color: #559cd5;">INFO</span>: Checking your credentials using rosie. Please wait.</terminal-line>
    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: Successfully accessed rosie with your credentials.</terminal-line>
</terminal-window>

### {{ model }} configuration
{{ model }} is comprised of 2 different suites: a [Regional Ancillary Suite (RAS)](#ras) and a [Regional Nesting Suite (RNS)](#rns).

Each {{ model }} suite has a `suite-ID` in the format `u-<suite-name>`, where `<suite-name>` is a unique identifier.<br>
Typically, an existing suite is copied and then edited as needed for a particular run.

> For more information on {{ model }} configuration, check [{{model}}][model configurations] page.

Author: Edit: "For more information on ACCESS-RAM, refer to the model configuration page."

    <terminal-line>[INFO] &lt;new-suite-ID&gt;: copied items from {{ ras_id }}/trunk@&lt;trunk-ID&gt;</terminal-line>
    <terminal-line>[INFO] {{ ras_id }}: local copy created at /home/565/&lt;$USER&gt;/roses/&lt;new-suite-ID&gt;</terminal-line>
</terminal-window>
When a new suite is created in this way, a _unique_ `<suite-ID>` is generated within the repository and populated with descriptive information about the suite and its initial configuration.

For additional `rosie` options, run:
```
rosie help
```

#### Run the suite
{{ model }} suites run on [_Gadi_](https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview) through a [PBS job] submission.<br>
When a suite runs, its configuration files are copied on _Gadi_ inside `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>` and a symbolic link to this directory is also created in the `$USER`'s home directory under `~/cylc-run/<suite-ID>`.<br>
{{ model }} suites comprise several tasks, such as checking out code repositories, compiling and building the different model components, running the model, etc. The workflow of these tasks is controlled by [_Cylc_](#cylc).

> To run the RAS run the following command from within the [suite directory](#suitedir):

Author: Edit: "To run the RAS, execute the following command from within..."

    <terminal-line>[INFO]  $ ps -opid,args &lt;PID&gt;  # on &lt;persistent-session-full-name&gt;</terminal-line>
    <img src="/assets/run_access_cm/Cylc_GUI_are.png" alt="Cylc GUI" imageTime="inf" loading="lazy">
</terminal-window>

TODO
{: style="color:red"}
<!-- Replace with a RAS image -->

- **HARD restart**<br>
    To overwrite any previous runs of the suite and start afresh, run the following command from within the [suite directory](#suitedir):
    ```
    rose suite-run --new
    ```

    !!! warning
        This will overwrite all existing model output and logs for the same suite.

##### RELOAD a suite
In some cases the suite needs to be updated without necessarily having to stop it (e.g., after fixing a typo in a file). Updating an active suite is called a _reload_, where the suite is _re-installed_ and _Cylc_ is updated with the changes. This is similar to a _SOFT_ restart, except new changes are installed, so you may need to manually trigger failed tasks from the _Cylc_ GUI.

To reload a suite, run the following command from within the [suite directory](#suitedir):
```
rose suite-run --reload
```

#### RAS output files
    

> The RAS output ancillary files can be found in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils`. The ancillaries are divided for each nested region, with each of the region subdivided in directories named according to the nests names. Therefore, the path of the ancillaries for a specific nest is `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/<nested_region_name>/<nest_name>`.

Author: Edit: "... nests' names."

<!--
TODO
{: style="color:red"}
<!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

> In addition to the chosen nests, a subdirectory named according to the driving model data can be found within the `nested_region_name` directory.<br>

Author: Edit: "In addition to these nests,..."

> The example above has one `nested_region_name` called `Lismore`, with a 2-level nest (2 nests) named `d1000` (outer nest) and `d0198` (inner nest), driven by ERA5 data.

Author: Edit: "The 2-level (i.e., two nests) example above has one `nested_region_name` called `Lismore`. Driven by ERA5 data, the outer and inner nests are named `d1000` and `d0198`, respectively, "

> As a consequence, within the ancillary files directory `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/` the following folders will be present:

Author: Edit: "Thus, the ancillary files directory `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/` contains the following folders:"

- `Lismore/d1000`
- `Lismore/d0198`
- `Lismore/era5`

The ancillary data files are typically in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

#### Edit the RAS configuration
This section describes how to modify the RAS configuration.<br>
The modifications discussed in this section can change the way the RAS is run and, as a result, control the ancillaries that are generated.

In general, ACCESS modelling suites can be edited either by directly modifying the configuration files within the [suite directory](#suitedir), or by using the [_Rose_ GUI](#rosegui).

!!! warning 

>     Unless you are a very expert user, directly modifying configuration files is usually discouraged to avoid incurring in errors.

Author: Edit: "Unless you are an experienced user, direct modification of configuration files is discouraged to avoid errors."

##### Rose GUI {: id="rosegui"}
To open the [_Rose_](#rose) GUI, run the following command from within the [suite directory](#suitedir): 
```
rose edit &
```

!!! tip
    The `&` is optional. It allows the terminal prompt to remain active while running the `Rose` GUI as a separate process in the background.
<!--
<terminal-window>
    <terminal-line data="input">cd ~/roses/&lt;suite-ID&gt;</terminal-line>
    <terminal-line data="input" directory="~/roses/&lt;suite-ID&gt;">rose edit &</terminal-line>
    <terminal-line class="ls-output-format">[&lt;N&gt;] &lt;PID&gt;</terminal-line>
    <terminal-line data="input" directory="~/roses/&lt;suite-ID&gt;"></terminal-line>
    <img src="/assets/run_access_cm/Rose_GUI_are.png" alt="Rose GUI" imageTime="inf" loading="lazy">
</terminal-window>

TODO
{: style="color:red"}
<!-- Change image with a RAS one -->

##### Change the region centre and name

> {{model}} is a regional model that can perform simulations for a specific region on Earth. 

Author: Edit: "In ACCESS-RAM, it is possible to change the name and centre of the specified region."

In the RAS, each region can be specified through the following parameters:

- *region centre* &rarr; `_rg01_centre_` within the suite.<br>
  Set of latitude/longitude coordinates defining the centre of the region.

- *nested region name* &rarr; `_rg01_name_` within the suite<br>
  Name for the nested region.

> In `{{ras_id}}` these prameters are set by default to the *Lismore Flood* example. 

Author: Typo: "parameters"

<!--
TODO
{: style="color:red"}
<!-- Can we add a link for the Lismore Flood example? Anything that makes it a bit clearer what that is. -->

To change these parameters, edit the `_rg01_name_` and `_rg01_centre_` fields in _suite conf &rarr; Regional Ancillary Suite &rarr; Nested region 1 setup_, and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

For example, to run an {{ model }} suite for a region with centre (-12.4, 130.8) and name "Darwin":

![Rose change region centre example](/assets/run_access_ras/ras_edit_region_center_and_name.gif){: class="example-img" loading="lazy"}

<!--
TODO
{: style="color:red"}
<!-- Add a gif/video for the exmple -->

### Regional Nesting Suite (RNS) {: id="rns"}
<!--
The RNS is ...
-->
The `suite-ID` of the RNS is `{{ rns_id }}`.

<!--
TODO
{: style="color:red"}
<!-- Add short description of what the RAS does -->

#### Get and run RNS configuration

> Steps to get and run the RNS configuration, as well as monitoring the runs, are analogous to those listed above for the [RAS](#ras).<br>

Author: Replace "get" with "obtain"

To monitor the RNS suite run, please follow the steps listed in [Monitor the suite runs](#monitor-the-suite-runs).

#### RNS output files

All the RNS output files are available on _Gadi_ inside `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>`. They are also symlinked in `~/cylc-run/<suite-ID>`.

The RNS output data can be found in the `/scratch/$PROJECT/$USER/cycl-run/<suite-ID>/share/cycle` directory, grouped for each [cycle](#change-run-length-and-cycling-frequency).<br>
Within the cycle directory, outputs are divided into multiple nested subdirectories in the format `<nested_region>/<science_choice>/<nest_name>`, with [nested_region](), [science_choice]() and [nest_name]() referring to the respective configurable options in the [RAS](#edit-the-ras-configuration).

<!--
TODO
{: style="color:red"}
<!-- Use the same names for `nested_region` `science_choice` and `nest_name` as the one used above for the RAS and make sure all of the are referenced in the RAS and there are clear instructions on how to modify them. There is no reference for how to change the `science_choice`. If it cannot be changed, I would call it in a different way. Add links -->

> Each of the `<nest_name>` directory has the following subdirectories:

Author: Edit: "Each `<nest_name>` directory..."

{% set model = "ACCESS-RAM3" %}
{% set ancillary_model = "ACCESS-RAS" %}
{% set regional_model = "ACCESS-RNS" %}

> {% set suite = "u-dg767" %}

Reviewer: "{% set suite = "u-dg767" %} should be no longer needed due to ras_id and rns_id being available.

Reviewer 2: Good point! 
Removed in [0e840f69e1dec91021fc50059b3e6a93f4fe0345](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/0e840f69e1dec91021fc50059b3e6a93f4fe0345)

```
echo "<name>.${USER}.<project>.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
```

For example, if the user `abc123` started a persistent session named `cylc` under the project `xy00`, the command will be:

<terminal-window data="input">
    <terminal-line>echo "cylc.abc123.xy00.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
    </terminal-line>
    <terminal-line data="input" linedelay="1000">cat ~/.persistent-sessions/cylc-session</terminal-line>
    <terminal-line data="output">cylc.abc123.xy00.ps.gadi.nci.org.au</terminal-line>
</terminal-window>

<!-- Updated example with echo command instead of cat
<!-- 
For example, if the user `abc123` started a persistent session named `cylc` under the project `xy00`, the command will be:
<terminal-window data="input">
    <terminal-line>cat > ~/.persistent-sessions/cylc-session <<< cylc.abc123.xy00.ps.gadi.nci.org.au</terminal-line>
    <terminal-line data="input" linedelay="1000">cat ~/.persistent-sessions/cylc-session</terminal-line>
    <terminal-line data="output">cylc.abc123.xy00.ps.gadi.nci.org.au</terminal-line>
</terminal-window>

> --->

Author: I updated the example with `echo` command instead of `cat` for simplicity (see comment above `<!-- Updated example with echo command instead of cat`). I then commented out the animation as it used `cat` and forgot to update and reinstate it. Sorry @atteggiani!

    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: Successfully accessed rosie with your credentials.</terminal-line>
</terminal-window>

### {{ model }} configuration
{{ model }} is comprised of 2 different suites: a [Regional Ancillary Suite (RAS)](#ras) and a [Regional Nesting Suite (RNS)](#rns).

Each {{ model }} suite has a `suite-ID` in the format `u-<suite-name>`, where `<suite-name>` is a unique identifier.<br>
Typically, an existing suite is copied and then edited as needed for a particular run.

For more information on {{ model }} configuration, check [{{model}}][model configurations] page.

!!! info 
    Many of the steps that follow are to be repeated almost identically for the RAS and RNS. For this reason, details on these steps will be provided only within the RAS section below, whereas in the following RNS section they will only be linked for reference.

### Regional Ancillary Suite (RAS) {: id="ras"}

The RAS produces a set of input (ancillary) files, which are then used by the RNS.

The `suite-ID` of the RAS is `{{ ras_id }}`.
<!--
TODO
{: style="color:red"}

> <!-- Add short description of what the RAS does -->

Reviewer: This is a bug that was created by putting the two separate mark-up files I had created into one.

In my branch the model and suite variables were defined at the top of the file.  The Markup then filled in the variables when the website was displayed.

The description of what the RAS and RNS did was also included in my original branch.  Please copy from there.

> <!-- Add short description of what the RAS does -->

Reviewer: The definition of the model and suite-id for RAS variables were defined here:

https://github.com/ACCESS-NRI/access-hive.org.au/blob/7a7f4efb17a3109eb63722bfd36f83574e507c5e/docs/models/run-a-model/run-access-r-ras.md?plain=1#L1

The definition of the model and suite-id for the RNS variables were defined here:

https://github.com/ACCESS-NRI/access-hive.org.au/blob/7a7f4efb17a3109eb63722bfd36f83574e507c5e/docs/models/run-a-model/run-access-r-rns.md

> <!-- Add short description of what the RAS does -->

Reviewer 2: Sorry, but I cannot find any description of the `RNS` in the pages that you linked.

Can you please add a short paragraph in a comment below, with a suitable description of the `RNS` that can be used in this portion of the documentation? Thank you.

> <!-- Add short description of what the RAS does -->

Reviewer: The suite-id was defined in the line underneath.
i.e.
https://github.com/ACCESS-NRI/access-hive.org.au/blob/7a7f4efb17a3109eb63722bfd36f83574e507c5e/docs/models/run-a-model/run-access-r-ras.md?plain=1#L2

> <!-- Add short description of what the RAS does -->

Reviewer 2: Sorry, my message was not clear.

The main issue here is not related to the RAS suite id. 
The content that is needed is a short introduction about the RAS (what it is, what it does, etc.).

Could you please provide that?

For additional `rosie` options, run:
```
rosie help
```

#### Run the suite
{{ model }} suites run on [_Gadi_](https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview) through a [PBS job] submission.<br>
When a suite runs, its configuration files are copied on _Gadi_ inside `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>` and a symbolic link to this directory is also created in the `$USER`'s home directory under `~/cylc-run/<suite-ID>`.<br>
{{ model }} suites comprise several tasks, such as checking out code repositories, compiling and building the different model components, running the model, etc. The workflow of these tasks is controlled by [_Cylc_](#cylc).

To run the RAS run the following command from within the [suite directory](#suitedir):
```
rose suite-run
```

After the initial tasks are executed, the _Cylc_ GUI will open. You can now view and control the different tasks in the suite as they are run. <!--<img src="/assets/run&UnderBar;access_cm/Cylc_GUI_are.png" alt="Cylc GUI" imageTime="inf" loading="lazy"> -->
<!--
TODO
{: style="color:red"}

> <!-- Replace image with RAS one -->

Reviewer 2: At the moment these animation will not be included.

You are done!

If you do not get any errors, you can check the suite output files after the run is complete.<br>
You can now close the _Cylc_ GUI. To open it again, run the following command from within the [suite directory](#suitedir):
```
rose suite-gcontrol
```

#### Monitor the suite runs

It is quite common, especially during the first few runs, to experience errors and job failures. Running a suite involves the execution of several tasks, and any of these tasks could fail. When a task fails, the suite is halted and a red icon appears next to the respective task name in the _Cylc_ GUI.<br>
To investigate the cause of a failure, we can look at the logs `job.err` and `job.out` from the suite run. There are two main ways to do so:

##### Using the _Cylc_ GUI
Right-click on the task that failed and click on _View Job Logs (Viewer) &rarr; job.err_ (or _job.out_).<br>
To access a specific task, click on the arrow next to the task to extend the drop-down menu with all the subtasks.

<!--
![Investigate Error GUI example](/assets/run_access_cm/investigate_error_gui_are.gif){: class="example-img" loading="lazy"}

TODO
{: style="color:red"}

> <!-- Replace with a RAS video/gif -->

Reviewer: It could be copied from /assets/run_access_cm/investigate_error_gui_are.gif but may need to made specific to this example.

<!-- 
TODO
{: style="color:red"}
<!-- Replace TASK_NAME with the actual task name according to the video/gif above -->

#### Stop, restart and reload suites
In some cases, you may want to control the running state of a suite.<br>
If your _Cylc_ GUI has been closed and you are unsure whether your suite is still running, you can scan for active suites and reopen the GUI if desired.<br>
To scan for active suites, run:
```
cylc scan
```
To reopen the _Cylc_ GUI, run the following command from within the [suite directory](#suitedir):
```
rose suite-gcontrol
```
<!--
<terminal-window>
    <terminal-line data="input">cylc scan</terminal-line>
    <terminal-line>&lt;suite-ID&gt; &lt;$USER&gt;@&lt;gadi-cpu&gt;.nci.org.au:&lt;port&gt;</terminal-line>
    <terminal-line data="input">cd ~/roses/&lt;suite-ID&gt;</terminal-line>
    <terminal-line data="input" directory="~/roses/&lt;suite-ID&gt;">rose suite-gcontrol</terminal-line>
    <img src="/assets/run_access_cm/Cylc_GUI_are.png" alt="Cylc GUI" imageTime="inf" loading="lazy">
</terminal-window>

TODO
{: style="color:red"}

> <!-- Replace with a RAS video/gif -->

Reviewer: It could be copied from /assets/run_access_cm/Cylc_GUI.png but it may still need to be edited.

    ```
    rose suite-run --new
    ```

    !!! warning
        This will overwrite all existing model output and logs for the same suite.

##### RELOAD a suite
In some cases the suite needs to be updated without necessarily having to stop it (e.g., after fixing a typo in a file). Updating an active suite is called a _reload_, where the suite is _re-installed_ and _Cylc_ is updated with the changes. This is similar to a _SOFT_ restart, except new changes are installed, so you may need to manually trigger failed tasks from the _Cylc_ GUI.

To reload a suite, run the following command from within the [suite directory](#suitedir):
```
rose suite-run --reload
```

#### RAS output files
    
The RAS output ancillary files can be found in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils`. The ancillaries are divided for each nested region, with each of the region subdivided in directories named according to the nests names. As a consequence, the path of the ancillaries for a specific nest is `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/<nested_region_name>/<nest_name>`.
<!--
TODO
{: style="color:red"}

> <!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

Reviewer: The instructions are generic.  The naming in the given example is specific to a given example and would most likely be immediately changed by the user.  i.e. most users will immediately change the example from Lismore rainfall to what they want.  It might be good if they can still refer to the run-a-model if they forget something later on.

> <!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

Reviewer: I don't really understand how the "generic" information is going to fit into the overview page.

By the overview page are you meaning something like: https://access-hive.org.au/models/configurations/access-esm/#model-components?

"What is nesting" doesn't seem to fit there.

> <!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

Reviewer 2: @chermelle The page you referenced is only an example to get started, but it can definitely be amplified and more paragraphs can be added depending on the specific pieces of information we want to provide.
A paragraph on "what is nesting" could be a suitable addition.

I will list below the pieces of information that I think is missing here and it might be important for a user:
- information about nesting (what it means and why is it necessary)
- information about the nesting structure within ACCESS-RAM (what can be changed related to nesting within the configuration? what are the variables that control those changes? How does that relate to the naming of outputs?

> <!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

Reviewer: As discussed in the team meeting today, I will add a brief outline but reference the UKMO RAL paper - specifically RAL1 for beginners to look at.

> <!-- There is no information before about nests (e.g., nest structure, what is the difference between nested region name and nest name, etc.) It would be good to add something in the suite description and link it here. -->

Reviewer 2: That sounds good. 
That paragraph will be placed in the "model overview" page and referenced (linked) here.

<!-- 
In the default two-level nest there is one nested region with three nests named era5, d1000 and d0198.  In this case the output directories:
Lismore/era5
Lismore/d1000
Lismore/d0198

TODO
{: style="color:red"}

> <!-- Why in a two-level nest there are three nests? This passage is not clear. -->

Reviewer: The era5 level nest is purely for the cutting out and reconfiguring of the ERA5 driving model data.  (It is need to run the setup but does not affect the number of nests used in the actual forecast).
The other two relate to the actual nests that a forecast/project will be run on.

> <!-- Why in a two-level nest there are three nests? This passage is not clear. -->

Reviewer: I would keep it simple as:

Lismore/era5  # the directory holding data cut out of from the (global) driving era5 data prior to reconfiguration onto the first level of the nest (slightly larger than the first level domain to create lateral boundary conditions as well as initial conditions from).

Lismore/d1000 # the directory holding data from first level of the nest (10km in this era5-land example).

Lismore/d0198 # the directory holding data from finest level nest.

> <!-- Why in a two-level nest there are three nests? This passage is not clear. -->

Reviewer 2: Changes added in [297fc9e0644f2f79556420e069369b4b5a462679](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/297fc9e0644f2f79556420e069369b4b5a462679)

<!--
The ancillary data files are typically in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) or namelist files formats.

TODO
{: style="color:red"}

> <!-- Are the namelists always produced? What are they produced for? Do we need to mention namelists as well here? Are the namelists Fortran namelists? In case I would specify it (if we decide to mention them). -->

Reviewer: This text was intended to just inform the users the format of the files produced by the ancillary suite.  The answer being that they are in UM FF format.  It not related to namelists.

In general, ACCESS modelling suites can be edited either by directly modifying the configuration files within the [suite directory](#suitedir), or by using the [_Rose_ GUI](#rosegui).

!!! warning 
    Unless you are a very expert user, directly modifying configuration files is usually discouraged to avoid incurring in errors.

##### Rose GUI {: id="rosegui"}
To open the [_Rose_](#rose) GUI, run the following command from within the [suite directory](#suitedir): 
```
rose edit &
```

!!! tip
    The `&` is optional. It allows the terminal prompt to remain active while running the `Rose` GUI as a separate process in the background.
<!--
<terminal-window>
    <terminal-line data="input">cd ~/roses/&lt;suite-ID&gt;</terminal-line>
    <terminal-line data="input" directory="~/roses/&lt;suite-ID&gt;">rose edit &</terminal-line>
    <terminal-line class="ls-output-format">[&lt;N&gt;] &lt;PID&gt;</terminal-line>
    <terminal-line data="input" directory="~/roses/&lt;suite-ID&gt;"></terminal-line>
    <img src="/assets/run_access_cm/Rose_GUI_are.png" alt="Rose GUI" imageTime="inf" loading="lazy">
</terminal-window>

TODO
{: style="color:red"}

> <!-- Change image with a RAS one -->

Reviewer: This could be copied from /assets/run_access_cm/Rose_GUI.png but may need to be made specific to this example.

##### Change the region centre and name
{{model}} is a regional model that can perform simulations for a specific region on Earth. 

In the RAS, each region can be specified through the following parameters:

- *region center* &rarr; `_rg01_centre_` within the suite 

<!--  
TODO
{: style="color:red"}

> <!-- Is this a set of coordinates? It would be good to be more specific here. -->

Reviewer: Yes.  The GUI requests the central latitude/longitude of the group of ancillaries being produced.  This part of the RAS instructions was help users change the central point of the domain.  A screenshot of the GUI may be more useful here but not strictly necessary.

> <!-- Is this a set of coordinates? It would be good to be more specific here. -->

Reviewer: I would have to see it marked up to comment.

> <!-- Is this a set of coordinates? It would be good to be more specific here. -->

Reviewer 2: Changed it in [508631c6e9ffa60ce26788a28b47cb8d4e9c1c85](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/508631c6e9ffa60ce26788a28b47cb8d4e9c1c85).

Feel free to view it through the PR preview and comment within the code for other suggestions.

- *nested region name* &rarr; `_rg01_name_` within the suite
<!--
TODO
{: style="color:red"}

> <!-- Are there any more parameters that can be set here? For example the extents of the different nests, etc... ? -->

Reviewer: This was a simple example.  
We can give other sets of instructions for modifying the number of pixels if required.
There are many details that can be changed.

In `{{ras_id}}` these prameters are set by default to the *Lismore Flood* example. 

<!--
TODO
{: style="color:red"}

> <!-- Can we add a link for the Lismore Flood example? Anything that makes it a bit clearer what that is. -->

Reviewer: We don't have any ACCESS-NRI produced material produced relating to the example.  It may make sense to create a separate website with the input/output explained.

To change these parameters, edit the `_rg01_name_` and `_rg01_centre_` fields in _suite conf &rarr; Regional Ancillary Suite &rarr; Nested region 1 setup_, and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

<!-- 
For example, to run an {{ model }} suite for a region with centre (-12.4, 130.8) and name "Darwin":

> -->

Reviewer: This was not commented out on my branch.

![Rose change region centre example](/assets/run_access_ras/ras_edit_region_center_and_name.gif){: class="example-img" loading="lazy"}

<!--
TODO
{: style="color:red"}

> <!-- Add a gif/video for the exmple -->

Reviewer: This asset exists in my /assets/run_access_ras/ras_edit_region_center_and_name.gif but please note the RAS not RAM naming.  It was from when I have divided the text into ancillaries and forecast/projection.

> <!-- Add a gif/video for the exmple -->

Author: This still needs work.

### Regional Nesting Suite (RNS) {: id="rns"}
<!--
The RNS is ...
-->
The `suite-ID` of the RNS is `{{ rns_id }}`.

<!--
TODO
{: style="color:red"}

> <!-- Add short description of what the RAS does -->

Reviewer: Those variables get filled in by the markup when it is run.

The description for the RNS was included in my original branch.

> <!-- Add short description of what the RAS does -->

Reviewer: The suite id was defined here:

https://github.com/ACCESS-NRI/access-hive.org.au/blob/7a7f4efb17a3109eb63722bfd36f83574e507c5e/docs/models/run-a-model/run-access-r-rns.md?plain=1#L2

and this short suitable description followed the ones from ACCESS-CM

https://github.com/ACCESS-NRI/access-hive.org.au/blob/7a7f4efb17a3109eb63722bfd36f83574e507c5e/docs/models/run-a-model/run-access-r-rns.md?plain=1#L150

> <!-- Add short description of what the RAS does -->

Reviewer 2: Sorry, my message was not clear.

The main issue here is not related to the RNS suite id.

The information about the components you referenced is good, but I would add a more general introduction about what the RNS is, what it does, etc.

#### Get and run RNS configuration
Steps to get and run the RNS configuration, as well as monitoring the runs, are analogous to those listed above for the [RAS](#ras).<br>
The only difference is the `suite-ID` that, for the RNS is `{{ rns_id }}`.

To get the RNS configuration, please follow the steps listed in [Get the RAS configuration](#get-the-ras-configuration), making sure you use the correct RNS `suite-ID` `{{ rns_id }}` when copying the suite.

To run the RNS configuration, please follow the steps listed in [Run the suite](#run-the-suite).

To monitor the RNS suite run, please follow the steps listed in [Monitor the suite runs](#monitor-the-suite-runs).

#### RNS output files

All the RNS output files are available on _Gadi_ inside `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>`. They are also symlinked in `~/cylc-run/<suite-ID>`.

> The RNS output data can be found in the `/scratch/$PROJECT/$USER/cycl-run/<suite-ID>/share/cycle` directory, grouped for each [cycle](#change-run-length-and-cycling-frequency).<br>
> Within the cycle directory, outputs are divided into multiple nested subdirectories in the format `<nested_region>/<science_choice>/<nest_name>`, with [nested_region](), [science_choice]() and [nest_name]() referring to the respective configurable options in the [RAS](#edit-the-ras-configuration).
> 
> <!--
> TODO
> {: style="color:red"}
> <!-- Use the same names for `nested_region` `science_choice` and `nest_name` as the one used above for the RAS and make sure all of the are referenced in the RAS and there are clear instructions on how to modify them. There is no reference for how to change the `science_choice`. If it cannot be changed, I would call it in a different way. Add links -->
> 
> Each of the `<nest_name>` directory has the following subdirectories:
> 
> - `ics` &rarr; initial conditions
> - `lbcs` &rarr; lateral boundary conditions
> - `um` &rarr; model output data
> 
> <!--
> TODO
> {: style="color:red"}
> <!-- Use the same names for `nested_region` `science_choice` and `nest_name` as the one used above for the RAS and make sure all of the are referenced in the RAS and there are clear instructions on how to modify them. Add links -->

Reviewer 2: ```suggestion
The RNS output data can be found in the `/scratch/$PROJECT/$USER/cycl-run/<suite-ID>/share/cycle` directory, grouped for each [cycle](#change-run-length-and-cycling-frequency).<br>
Within the cycle directory, outputs are divided into multiple nested subdirectories in the format `<nested_region>/<science_choice>/<nest_name>`, with [nested_region](<LINK>), [science_choice](<LINK>) and [nest_name](<LINK>) referring to the respective configurable options in the [RAS](#edit-the-ras-configuration).

Each of the `<nest_name>` directory has the following subdirectories:

- `ics` &rarr; initial conditions
- `lbcs` &rarr; lateral boundary conditions
- `um` &rarr; model output data
```

Use the same names for `<nested_region>` `<science_choice>` and `<nest_name>` as the one used above for the RAS and make sure all of them are referenced in the RAS and there are clear instructions on how to modify them. Make sure `<LINK>` is substituted with the proper link to the related portion/paragraph of the RAS.

There is no reference to how to change the `science_choice`, it would be good to add instructions. If it cannot be changed, I would maybe call it in a different way.

Reviewer: By default the RNS puts data into directories named with a convention naming the science choice.

Changing the science choice has implications beyond a "introductory simple guide".  Perhaps we should note that it is out of scope for the "run-a-model" guide.

The model output data for the `Lismore` `nested_region`, using a `RAL3P2` `science_choice` and `d0198` as a `nest_name`, will be stored in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/cycle/20220226T0000Z/Lismore/d0198/RAL3P2/um`.

The RNS atmospheric output data files are typically in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

#### Edit the RNS configuration

This section describes how to modify the RNS configuration.
The modifications discussed in this section can change the way the RNS is run, or how its specific [model components] are configured and coupled together.

In general, ACCESS modelling suites can be edited either by directly modifying the configuration files within the suite directory, or by using the [_Rose_ GUI](#rosegui).

!!! warning
    Unless you are a very expert user, directly modifying configuration files is usually discouraged to avoid incurring in errors.

To open the _Rose_ GUI, you can follow the steps listed in [Rose GUI](#rosegui).

> #### Change where the output from run is stored project
> To change where the output from the model run is stored, edit the _NCI_STORAGE_ field in _suite conf &rarr; Nesting Suite &rarr; General run options _, and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}. 
> 
> <!--
> TODO
> {: style="color:red"}
> <!-- I couldn't restructure this title/text because from them I am not really understanding what `NCI_STORAGE` is. From its default value in the rose-suite.conf (scratch/$PROJECT) and how it's used inside the suite, it seems to be a project folder added to the `storage` option for the PBS jobs submission of many tasks. Therefore is not necessarily related to model outputs (is also related to model input), but to filesystem mounts that the model need to be able to access. Why is the text mentioning changing where the output is stored? -->

Reviewer 2: ```suggestion
#### Change where the output from run is stored project
To change where the output from the model run is stored, edit the _NCI_STORAGE_ field in _suite conf &rarr; Nesting Suite &rarr; General run options _, and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}. 
```
I am not sure what `NCI_STORAGE` is. From its default value in the `rose-suite.conf` (`scratch/$PROJECT`) and how it's used inside the suite, it seems to be a project folder added to the `storage` option for the PBS jobs submission of multiple tasks. Therefore, is not necessarily related to model outputs but also related to model input. 
 I would say in general is related to filesystem mounts that the model need to be able to access.
 Why is the text mentioning changing where the output is stored?

Reviewer: The NCI_STORAGE impacts where the user wants the output of their runs to be stored.
It is mentioned to given an example of changing where the output from the run is stored.

Reviewer 2: Looking at the RNS (`u-dg768`) I can only see `NCI_STORAGE` being added to the `pbs` `storage` option for different tasks, but I cannot see any part that link it to the experiment output.
I can see `NCI_STORAGE_TOP_DIR` used to provide an env variable to the `ARCHIVE` task (which might define where the output is being stored).

Can you please point me to a portion of the RNS suite where there is link between the `NCI_STORAGE` variable and the output of the run?
Thank you

Reviewer: The NCI_STORAGE refers to the User Interface of the rose gui.

Reviewer 2: The User Interface refers to variables used within the suite configuration files.

In the Rose GUI metadata description file `rose-meta.conf`, `NCI_STORAGE` is described as:
> Extra storage mounts (e.g. "gdata/a12+scratch/cd3")
> Additionally gdata/access will always be added

There is no mention to the outputs.

Based on the `rose-meta.conf` file and the `suite-adds.rc` files, `NCI_STORAGE` doesn't seem to be connected to any output.
A variable that might be connected to the output archiviation location is, instead, `NCI_STORAGE_TOP_DIR`.

#### Change run length and cycling frequency
The RNS is run in multiple steps, each one constituting a cycle. The job scheduler resubmits the suite every chosen cycling frequency, until the total run length is reached.

For the RNS:

- `CYCLE_INT_HR` is the cycling frequency (in hours)
- The total run length is the time span between the `INITIAL_CYCLE_POINT` and `FINAL_CYCLE_POINT`

> To modify these parameters, navigate to _suite conf &rarr; Nesting Suite &rarr; Cycling options_, edit the `INITIAL_CYCLE_POINT`, `FINAL_CYCLE_POINT` and the `CYCLE_INT_HR` fields (using [ISO 8601 Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations) format) and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.
> 
> <!--
> TODO
> {: style="color:red"}
> <!-- Are there any other options more specific to the RNS that users may want to change here? -->

Reviewer 2: ```suggestion
To modify these parameters, navigate to _suite conf &rarr; Nesting Suite &rarr; Cycling options_, edit the `INITIAL_CYCLE_POINT`, `FINAL_CYCLE_POINT` and the `CYCLE_INT_HR` fields (using [ISO 8601 Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations) format) and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.
```
Are there any further options specific to the RNS, that users may want to change, that could be added here?

Reviewer 2: @chermelle 
This is related to another comment above, about the configuration options that we want to include in the instructions.
I think whatever configuration options we decide to detail in the instructions, we might want to add the options specific to the RNS here.

Reviewer: I think this is best handled by showing a screenshot of the rose UI or just letting the user view it if there are licensing issues.

> <!--
> ## Get Help
> If you have questions or need help regarding {{ model }}, consider creating a topic in the [Earth System Model category of the ACCESS-Hive Forum](https://forum.access-hive.org.au/c/esm/earth-system-model/72).<br>
> For assistance on how to request help from ACCESS-NRI, follow the [guidelines on how to get help](/about/user_support/#still-need-help).
> 
> TODO
> {: style="color:red"}
> <!-- Check and change link details if needed -->

Reviewer 2: ```suggestion
## Get Help
If you have questions or need help regarding {{ model }}, consider creating a topic in the ........ category of the ACCESS-Hive Forum](<LINK>).<br>
For assistance on how to request help from ACCESS-NRI, follow the [guidelines on how to get help](/about/user_support/#still-need-help).
```
Add the right forum category with the `<LINK>`.

Reviewer 2: @ACCESS-NRI/webops can take care of this, once there is a Forum post of the release.

{% set model = "ACCESS-RAM3" %}
{% set suite = "u-dg767" %}
{% set ras_id = "u-dg767" %}
{% set rns_id = "u-dg768" %}
[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[model components]: /models/configurations/access-ram/#model-components
[model configurations]: /models/configurations/access-ram

[:fontawesome-brands-github:{: class="twemoji icon-before-text"} {{ model }} configuration]({{github_configs}}){: class="text-card"}
<!--
TODO
{: style="color:red"}

> <!-- Can we put a configuration here taken from MOSRS instead of GitHub? SVN? -->

Reviewer: They would need to be two separate links to MOSRS.  The ones listed would be the right ones.

> <!-- Can we put a configuration here taken from MOSRS instead of GitHub? SVN? -->

Reviewer 2: Ok I'm going to add those links. Thank you!

> <!-- Can we put a configuration here taken from MOSRS instead of GitHub? SVN? -->

Reviewer 2: Added in [b058fa7686af11e2f9d953b5ad03a3f5398e3fc5](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/b058fa7686af11e2f9d953b5ad03a3f5398e3fc5).

# Run ACCESS-RAM3

!!! info
    {{ model }} is a place-holder name until the final release name is chosen.

## About

{{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office](https://www.metoffice.gov.uk/) Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution.<br>
A description of the model and its components is available in the [ACCESS-RAM3 overview][model configurations].

> {{ model }} comprises a [Regional Ancillary Suite (RAS)](#ras) (RAS) `{{ ras_id }}` to generate ancillary files needed to run the [Regional Nesting Suite (RNS)](#rns) `{{ rns_id }}` for the domain of interest.

Author: Edit: remove duplicate "(RAS)"

> The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

Reviewer 2: @engelca any thoughts on this?

> The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

Reviewer: The python scripts are coming from the ACCESS-NRI software deployment pipeline but they are a component of the suite.  I would change the wording to "using ACCESS-NRI's supported configuration specifically designed to run on ..."

> The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

Reviewer 2: Changed in [8c045bb4fabb160d045af6384ac9d49b7444209c](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/8c045bb4fabb160d045af6384ac9d49b7444209c)

{% set model = "ACCESS-RAM3" %}
{% set suite = "u-dg767" %}
{% set ras_id = "u-dg767" %}
{% set rns_id = "u-dg768" %}
[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[model components]: /models/configurations/access-ram/#model-components
[model configurations]: /models/configurations/access-ram

[:fontawesome-brands-github:{: class="twemoji icon-before-text"} {{ model }} configuration]({{github_configs}}){: class="text-card"}
<!--
TODO
{: style="color:red"}
<!-- Can we put a configuration here taken from MOSRS instead of GitHub? SVN? -->

# Run ACCESS-RAM3

!!! info
    {{ model }} is a place-holder name until the final release name is chosen.

## About

> {{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office](https://www.metoffice.gov.uk/) Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution.<br>

Reviewer: I think that some people may not read the overview page.  If you want to move all the information there it would be worthwhile reminding them to look at the overview page for specific details.
In a bigger picture though these details are specific to the Lismore example not to using the RAS/RNS as users could change those options.

> {{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office](https://www.metoffice.gov.uk/) Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution.<br>

Reviewer: The nesting strategy outlined here (based on the Lismore example) only applies to that example.  Users may quickly deviate from that setup.

The information needs to go somewhere but it may not fit in the overview page (if by overview you mean something like https://access-hive.org.au/models/configurations/access-esm/#model-components).

> {{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office](https://www.metoffice.gov.uk/) Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution.<br>

Reviewer 2: If this information is only relevant to the specific example used in the `How to run page` (Lismore experiment), then I would add it within the About section, but I would move it after the paragraph:

> The instructions below outline how to run {{ model }} using ACCESS-NRI's supported configuration, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

We could then spend a few more words about the "instructions below", mentioning the specific experiment (Lismore) that is going to be referenced and giving a few more details about its configuration. 

What about:

... specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].<br>
As an example, an experiment modelling the [Lismore flood event](<link to the Lismore flood event>) will be run, having the following configuration:
- initial and lateral boundary conditions driven by ERA5 data
- 2-level [nest](<link to the "nesting" paragraph within the model overview page>):
  - An outer nest that uses GAL9 (n.d.r would be good to provide a link to GAL9, because we have never mentioned it here) with a 10km horizontal resolution.
  - An inner nest that focuses on the region of interest using RAL3.2 (n.d.r. same with GAL9, I would provide a link here) with a 2.2 km horizontal resolution.

A description of the model and its components is available in the [ACCESS-RAM3 overview][model configurations].

{{ model }} comprises a [Regional Ancillary Suite (RAS)](#ras) (RAS) `{{ ras_id }}` to generate ancillary files needed to run the [Regional Nesting Suite (RNS)](#rns) `{{ rns_id }}` for the domain of interest.

The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].

If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

> All {{model}} configurations are open source, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1")![CC icon](https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"}![BY icon](https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"} and available on [ACCESS-NRI GitHub]({{github_configs}}).

Reviewer 2: @engelca @aidanheerdegen 
Any information about the License?

> All {{model}} configurations are open source, licensed under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1")![CC icon](https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"}![BY icon](https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1){: style="height:1em;margin-left:0.2em;vertical-align:text-top;"} and available on [ACCESS-NRI GitHub]({{github_configs}}).

Reviewer 3: For any of the UKMO licensed suites this isn't correct. They're not open and they're not stored on GitHub. Is there any licensing information in the suites or on MOSRS?

[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[model components]: /models/configurations/access-ram/#model-components
[gadi]: https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview
[default project]: /getting_started/set_up_nci_account#change-default-project-on-gadi

<div class="text-card-group" markdown>

[![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} RAS configuration]({{mosrs_config_ras}}){: class="text-card" style=""}
[![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} RNS configuration]({{mosrs_config_rns}}){: class="text-card"}
[:notepad_spiral:{: class="twemoji icon-before-text"} Release notes]({{release_notes}}){: class="text-card"}
</div>

# Run {{ model }}

## About

{{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office (UKMO)](https://www.metoffice.gov.uk/) Regional Nesting Suite for high-resolution regional atmosphere modelling.<br>
A description of the model and its components is available in the [{{ model }} overview]({{ model_configurations }}/#{{ model }}).

> {{ model }} comprises two suites: a [Regional Ancillary Suite (RAS)](#ras), which generates ancillary files (i.e.: input files), and a [Regional Nesting Suite (RNS)](#rns) which runs the regional forecast.

Author: edit: change to "... which generates ancillary (input) files, and a ..."

The instructions below outline how to run {{ model }} using ACCESS-NRI's supported configuration, specifically designed to run on the [National Computating Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].<br>

> As an example, an experiment modelling a flood event in Lismore (NSW) will be run, using `ERA5-Land` [land-surface initial conditions]({{ model_configurations }}/#land-surface-initial-conditions-source) and having the configuration specified in [Nesting configuration]({{ model_configurations }}/#nesting-configuration).

Author: edit: `The example experiment focuses on a flood event in Lismore, NSW, using `ERA5-Land` [land-surface initial conditions]({{ model_configurations }}/#land-surface-initial-conditions-source), where the configuration is specified in the [Nesting configuration]({{ model_configurations }}/#nesting-configuration).`

If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

> All {{model}} configurations are and available on MOSRS, through the links at the top of this page.

Author: edit: `All {{model}} configurations are and available on MOSRS via links at the top of this page.`

    !!! tip
        To request membership for the _ki32_mosrs_ subproject, you need to:
        
        - already be member of the _ki32_ project
        {: style="list-style-type: disc"}
        - have a [MOSRS account](#mosrs-account)
        {: style="list-style-type: disc"}

    For more information on joining specific NCI projects, refer to [How to connect to a project](https://opus.nci.org.au/display/Help/How+to+connect+to+a+project).

- **Connection to an ARE VDI Desktop (optional)**<br>
    To run {{ model }}, start an [Australian Research Environment (ARE) VDI Desktop](https://are.nci.org.au/pun/sys/dashboard/batch_connect/sys/desktop_vnc/ncigadi/session_contexts/new) session.<br>
    If you are not familiar with ARE, check out the [Getting Started on ARE](/getting_started/are) section.

!!! warning
    Please note that the waiting time for some of the prerequisites mentioned above could take 2-3 weeks.

## Quick Start guide

The following *Quick Start* guide is aimed at experienced users wanting to run {{ model }}. For more detailed instructions, please refer to the [Detailed guide](#detailed-guide).

### Required setup for running {{ model }} {: .no-toc }

> - **Start a new [_persistent session_](https://opus.nci.org.au/display/Help/Persistent+Sessions)**<br> 

Author: I would remove the line break `<br>` mid sentence.

    using a *Gadi* login node or an ARE terminal instance:
    ```
    persistent-sessions start <name>
    ```
    This will use your [default project].

    For further instructions on starting a _persistent session_, refer to the [detailed guide](#start-a-new-persistent-session).

- **Assign the _persistent session_ to Rose/Cylc workflows**<br>
    Run the following command:
    ```
    echo "<name>.${USER}.<project>.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
    ```
    substituting `<name>` with the name given to your _persistent session_, and `<project>` with the project assigned to it.

    !!! tip
        This step should only be done once

>     For further instructions on assigning the target _persistent session_, refer to the [detailed guide](#specify-target-persistent-session).

Author: edit: Be consistent with`Detailed` of `detailed` guide. Both instances are used throughout. I would go with the proper noun `Detailed guide`

To get the _Cylc_ setup required to run {{ model }}, execute the following commands:
```
module use /g/data/hr22/modulefiles
module load cylc7
```
<terminal-window data="input">
    <terminal-line>module use /g/data/hr22/modulefiles</terminal-line>
    <terminal-line>module load cylc7</terminal-line>
    <terminal-line data="output">Using the cylc session &lt;name&gt;.&lt;$USER&gt;.&lt;project&gt;.ps.gadi.nci.org.au</terminal-line>
    <terminal-line data="output"></terminal-line>
    <terminal-line data="output">Loading cylc7/24.03</terminal-line>
    <terminal-line data="output">&emsp;Loading requirement: mosrs-setup/2.0.1</terminal-line>
</terminal-window>

!!! warning
    _Cylc_ version >= `cylc7/24.03` required.<br>
    
    Also, before loading the _Cylc_ module, make sure to have started a _persistent session_ and have assigned it to the {{ model }} workflow. For more information about these steps, refer to [Set up _persistent session_](#set-up-persistent-session).

#### Rose setup {: #rose .no-toc }
[Rose](http://metomi.github.io/rose/doc/html/index.html) is a toolkit that can be used to view, edit, or run an ACCESS modelling suite.

> By completing the [_Cylc_ setup](#cylc), also _Rose_ will be automatically available. Therefore, there is no additional step required.

Author: edit: `By completing the [_Cylc_ setup](#cylc), _Rose_ will also be automatically available. Hence, no additional step required.`

#### MOSRS authentication {: .no-toc }
To authenticate using your _MOSRS_ credentials, run:
```
mosrs-auth
```
<terminal-window>
    <terminal-line data="input">mosrs-auth</terminal-line>
    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: You need to enter your MOSRS credentials here so that GPG can cache your password.</terminal-line>
    <terminal-line>Please enter the MOSRS password for &lt;MOSRS-username&gt;:</terminal-line>
    <terminal-line lineDelay=1500>Checking your credentials using Subversion. Please wait.</terminal-line>
    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: Successfully accessed Subversion with your credentials.</terminal-line>
    <terminal-line lineDelay=100><span style="color: #559cd5;">INFO</span>: Checking your credentials using rosie. Please wait.</terminal-line>
    <terminal-line lineDelay=500><span style="color: #559cd5;">INFO</span>: Successfully accessed rosie with your credentials.</terminal-line>
</terminal-window>

!!! warning

>     This step needs to be done once for each new session (e.g.: _Gadi_ login, _ARE_ terminal window)

Author: edit: `... each new session (e.g., _Gadi_ login, _ARE_ terminal window)`

### {{ model }} configuration

> {{ model }} is comprised of 2 different suites: a [Regional Ancillary Suite (RAS)](#ras) and a [Regional Nesting Suite (RNS)](#rns).

Author: edit: `... {{ model }} comprises two different suites: ...`

> Each {{ model }} suite has a `suite-ID` in the format `u-<suite-name>`, where `<suite-name>` is a unique identifier.<br>

Author: edit: `Both of the suites within {{ model }} have a `suite-ID` in the format ...`

> Typically, an existing suite is copied and then edited as needed for a particular run.

Author: edit: replace `run` with `experiement'

For more information on {{ model }}, refer to the [{{model}} configuration]({{ model_configurations }}/#{{ model }}) page.

!!! info 

>     Many of the steps that follow are to be repeated almost identically for the RAS and RNS. For this reason, details on these steps will be provided only within the RAS section below, whereas in the following RNS section they will only be linked for reference.

Author: edit: `Many of the following steps appear in both the RAS and RNS. For this reason, these steps will be detailed only within the RAS section below, and subsequenltly linked to within the RNS section.`

### Regional Ancillary Suite (RAS) {: #ras }

> The RAS generates a set of ancillary files, such as initial conditions, lateral boundary conditions and forcing conditions, for the domain of interest. These ancillary files are then used by the [RNS](#rns).

Author: edit: `For the domain of interest, the RAS generates a set of ancillary files, such as initial conditions, lateral boundary conditions and forcing conditions.`

The `suite-ID` of the RAS is `{{ ras_id }}`.

#### Get the RAS configuration
[Rosie](https://metomi.github.io/rose/doc/html/tutorial/rose/furthertopics/rosie) is an [SVN](https://subversion.apache.org) repository wrapper with a set of options specific for ACCESS modelling suites. It is automatically available within the [_Rose_ setup](#rose).

> The RAS configuration can be obtained as a copy of the MOSRS one, following 2 approaches:

Author: edit: `The RAS configuration can be copied from the MOSRS repository via two ways:`

- [Local-only copy](#local-copy)
- [Remote and local copy](#remote-copy)

Suites are, by default, created in the user's home directory on _Gadi_ under `~/roses/<suite-ID>`.
This path will be referred to as the *suite directory*.
{: #suitedir }

The suite directory contains multiple subdirectories and files, including:

- `app` &rarr; directory containing the configuration files for various tasks within the suite.
- `meta` &rarr; directory containing the GUI metadata.
- `rose-suite.conf` &rarr; main suite configuration file.
- `rose-suite.info` &rarr; suite information file.
- `suite.rc` &rarr; _Cylc_ control script file (Jinja2 language).

##### Local-only copy {: #local-copy .no-toc }

> To create a _local copy_ of the RAS from the UKMO repository, run:

Author: edit: replace `UKMO` with `MOSRS'

```
rosie checkout {{ ras_id }}
```
<terminal-window>
    <terminal-line data="input">rosie checkout {{ ras_id }}</terminal-line>
    <terminal-line>[INFO] create: /home/565/&lt;$USER&gt;/roses</terminal-line>
    <terminal-line>[INFO] &lt;suite-ID&gt;: local copy created at /home/565/&lt;$USER&gt;/roses/{{ ras_id }}</terminal-line>
</terminal-window>

> This option is mostly used for testing and examining existing suites.

Author: remove `exisiting` - it's redundant

    
##### Remote and local copy {: #remote-copy .no-toc }

> To create a new copy of the RAS both _locally_ and _remotely_ in the UKMO repository, run: 

Author: edit: replace `UKMO` with `MOSRS`

```
rosie copy {{ ras_id }}
```
<terminal-window>
    <terminal-line data="input">rosie copy {{ ras_id }}</terminal-line>
    <terminal-line>Copy "{{ ras_id }}/trunk@&lt;trunk-ID&gt;" to "u-?????"? [y or n (default)]</terminal-line> <terminal-line data="input">y</terminal-line>
    <terminal-line>[INFO] &lt;new-suite-ID&gt;: created at https://code.metoffice.gov.uk/svn/roses-u/&lt;suite-n/a/m/e/&gt;</terminal-line>
    <terminal-line>[INFO] &lt;new-suite-ID&gt;: copied items from {{ ras_id }}/trunk@&lt;trunk-ID&gt;</terminal-line>
    <terminal-line>[INFO] {{ ras_id }}: local copy created at /home/565/&lt;$USER&gt;/roses/&lt;new-suite-ID&gt;</terminal-line>
</terminal-window>

> When a new suite is created in this way, a _unique_ `<suite-ID>` is generated within the repository and populated with descriptive information about the suite and its initial configuration.

Author: edit: `When a new suite is created in this way, a _unique_ `<suite-ID>` is generated within the MOSRS repository, where you are prompted to add information about the suite and its configuration.`

For additional `rosie` options, run:
```
rosie help
```

#### Run the RAS
{{ model }} suites run on [_Gadi_](https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview) through a [PBS job] submission.<br>

> When a suite runs, its configuration files are copied on _Gadi_ inside `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>` and a symbolic link to this directory is also created in the `$USER`'s home directory under `~/cylc-run/<suite-ID>`.<br>

Author: edit: `When a suite runs, its configuration files are copied in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>` on _Gadi_. A symbolic link to this directory is also created in the `$USER`'s home directory under `~/cylc-run/<suite-ID>`.`

{{ model }} suites comprise several tasks, such as checking out code repositories, compiling and building the different model components, running the model, etc. The workflow of these tasks is controlled by [_Cylc_](#cylc).

> To run the RAS, execute the following command from within the [suite directory](#suitedir):

Author: edit: `To run the RAS, execute the following command from within your RAS [suite directory](#suitedir):`

```
rose suite-run
```

After the initial tasks are executed, the _Cylc_ GUI will open, where it is possible to view and control the different tasks in the suite as they are run.

!!! tip
    The _Cylc_ GUI can be safely closed without impacting the experiment run.<br>
    To open it again, run the following command from within the [suite directory](#suitedir):
```
rose suite-gcontrol
```

All steps are completed!! <br>
You will be able to check the [suite output files](#ras-output-files) after the run successfully completes.<br>
If you get errors or you can't find the outputs, [check the suite logs](#check-suite-logs) for debugging.

#### Check suite logs

> It is quite common, especially at the beginning, to experience errors and job failures.<br>

Author: edit:`It is not unusual that new users experience errors and job failures.`

When a suite task fails, a red icon appears next to the respective task name in the _Cylc_ GUI.<br>

To investigate the cause of a failure, or to monitor the progress of a suite, it can be useful to look at the suite logs.

> Suite logs are stored in the directory `~/cylc-run/<suite-ID>` within a folder named `log.<TIMESTAMP>`, which is also symlinked as `log` (referred to as _logs folder_ below). Logs from previous runs of the same suite are archived as compressed files with the naming pattern `log.<TIMESTAMP>.tar.gz`.

Author: edit: `These files can be found in the...`

Inside the logs folder, various files and subfolders can be found. The most relevant logs are typically:

- [Suite execution log](#suite-execution-log)
- [Task-specific logs](#task-specific-logs)

##### Suite execution log {: #suite-execution-log .no-toc }

The primary suite execution log is located at: `~/cylc-run/<suite-ID>/log/suite/log`.

This file contains a chronological record of the suite's run history. Each line is a distinct log entry, generally formatted as: `<TIMESTAMP> <LOG-TYPE> - [<task-name>.<cylc-cycle-point>] <status>`.

> ??? code "Example of a suite execution log file (click to enlarge)"

Author: edit: `... (click to see contents)`

    2025-03-14T05:06:18Z INFO - [housekeep.01000301T0000Z] status=submitted: (received)started at 2025-03-14T05:06:16Z for job(01)
    2025-03-14T05:06:18Z INFO - [housekeep.01000301T0000Z] -health check settings: execution timeout=PT12H
    2025-03-14T05:06:20Z INFO - [housekeep.01000301T0000Z] status=running: (received)succeeded at 2025-03-14T05:06:19Z for job(01)
    2025-03-14T05:06:20Z INFO - Suite shutting down - AUTOMATIC
    2025-03-14T05:06:28Z INFO - DONE
    ``` 

This file helps identify specific tasks that failed during the suite run.

!!! tip
    When a task fails, the `LOG-TYPE` will typically be `ERROR` or `CRITICAL`, instead of the more common `INFO`.

Once a specific task and _Cylc_ cycle point are identified, the task-specific logs can be inspected.

##### Task-specific logs {: .no-toc }

Logs for individual tasks are located in subfolders within the logs folder, following this path structure:
```
~/cylc-run/<suite-ID>/log/job/<cylc-cycle-point>/<task-name>/<retry-number>
```
The `<retry-number>` indicates the number of retries for the same task, with the latest retry symlinked to `NN`.

> For example, the logs for the latest retry of a task named `Lismore_d1000_GAL9_um_recon` at _Cylc_ cycle point `20220226T0000Z` can be in the following folder:

Author: edit: `For example, logs for the most recent retry of a task named `Lismore_d1000_GAL9_um_recon` at Cylc cycle point `20220226T0000Z` can be found in the folder...`

```
~/cylc-run/<suite-ID>/log/job/20220226T0000Z/Lismore_d1000_GAL9_um_recon/NN
```

Within this directory, the `job.out` and `job.err` files (representing `STDOUT` and `STDERR`, respectively) can be found, along with other related log files.

!!! tip
    Within the _Cylc_ GUI, logs for a specific task can be viewed by right-clicking on the task and selecting the desired log from the _View Job Logs (Viewer)_ menu.

#### Stop, restart, reload and clean suites
In some cases, you may want to control the running state of a suite.<br>

> If your _Cylc_ GUI has been closed and you are unsure whether your suite is still running, you can scan for active suites and reopen the GUI if desired.<br>

Author: remove `if desired`

```
rose suite-gcontrol
```

##### STOP a suite {: .no-toc }
To shutdown a suite in a safe manner, run the following command from within the [suite directory](#suitedir):
```
rose suite-stop -y
```
Alternatively, you can directly kill the [PBS jobs][PBS job] connected to your run. To do so:

1. Check the status of all your PBS jobs:
```
qstat -u $USER
```

1. Delete any job related to your run:
```
qdel <job-ID>
```

##### RESTART a suite {: .no-toc }
There are two main ways to restart a suite:

- **_SOFT_ restart**<br>
    To reinstall the suite and reopen _Cylc_ in the same state it was prior to being stopped, run the following command from within the [suite directory](#suitedir):
    ```
    rose suite-run --restart
    ```

    !!! warning
        You may need to manually trigger failed tasks from the _Cylc_ GUI.

- **HARD restart**<br>
    To overwrite any previous runs of the suite and start afresh, run the following command from within the [suite directory](#suitedir):
    ```
    rose suite-run --new
    ```

    !!! warning
        This will overwrite all existing model output and logs for the same suite.

##### RELOAD a suite {: .no-toc }

> In some cases the suite needs to be updated without necessarily having to stop it (e.g., after fixing a typo in a file). Updating an active suite is called a _reload_, where the suite is _re-installed_ and _Cylc_ is updated with the changes. This is similar to a _SOFT_ restart, except new changes are installed, so you may need to manually trigger failed tasks from the _Cylc_ GUI.

Author: add comma: `In some cases, the...`

To reload a suite, run the following command from within the [suite directory](#suitedir):
```
rose suite-run --reload
```

##### CLEAN a suite {: .no-toc }
To remove all files and folders created by the suite within the `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>` directory, run the following command from within the [suite directory](#suitedir):
```
rose suite-clean
```

Alternatively, you can achieve the same behaviour within a new submission of an experiment, by appending the `--new` option to the `rose suite-run` command:
```
rose suite-run --new
```

!!! warning

>     Cleaning a suite folder will remove any non-archived data (i.e.: output files, logs, executables, etc.) associated with the suite.

Author: edit: `i.e., ` not `i.e.:`

#### RAS output files

The RAS output ancillary files can be found in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils`.<br>

> The ancillaries are divided for each [nested region]({{ model_configurations }}/#nesting), with each of the region subdivided in directories named according to the nests' names. Therefore, the path of the ancillaries for a specific nest is `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/<nested_region_name>/<nest_name>`.

Author: edit: `Ancillaries within a [nested region](https://access-hive.org.au/pr-preview/pr-857/models/configurations/access-ram/#nesting) are separated according to the nest resolution, which are stored in a subdirectories according to the name given to the nest resolution. Thus, the path of ancillaries for a specific nest is...`

> In addition to these nests, a subdirectory named according to the driving model data can be found within the `nested_region_name` directory.<br>

Author: edit: `Similarly, driving model data for each nest resolution can be found within the `nested_region_name` directory.`

> The 2-level (i.e., two nests) example above has one `nested_region_name` called `Lismore`. Driven by ERA5 data, the outer and inner nests are named `d1000` and `d0198`, respectively.<br>

Author: edit: `For the above 2-level nest example, there is one `nested_region_name` called `Lismore`. Driven by ERA5 data, the outer and inner nests (i.e., nest resolution 1 and nest resolution 2) are named `d1000` and `d0198`, respectively.`

> Thus, the ancillary files directory `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/` contains the following folders:

Author: replace 'folders' with 'subdirectories'

- `Lismore/d1000`
- `Lismore/d0198`
- `Lismore/era5`

> The ancillary data files in output are typically in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

Author: edit: `Ancillary data files are typically output in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.`

### Regional Nesting Suite (RNS) {: #rns }

The RNS uses the ancillary files produced by the RAS to run the regional forecast for the domain of interest.

The `suite-ID` of the RNS is `{{ rns_id }}`.

#### Get and run RNS configuration

> Steps to obtain and run the RNS configuration, as well as monitoring the logs, are analogous to those listed above for the [RAS](#ras).<br>

Author: edit: `Steps to obtain and run the RNS suite, as well as monitor logs, are similar to those listed above for the [RAS](#ras).<br>`

The only difference is the `suite-ID`, which for the RNS is `{{ rns_id }}`.

> To get the RNS configuration, please follow the steps listed in [Get the RAS configuration](#get-the-ras-configuration), making sure you use the correct RNS `suite-ID` `{{ rns_id }}` when copying the suite.

Author: remove `please`

> To run the RNS configuration, please follow the steps listed in [Run the suite](#run-the-ras).

Author: remove `please`

> To check the RNS suite logs, please follow the steps listed in [Check suite logs](#check-suite-logs).

Author: remove `please`

#### RNS output files

> All the RNS output files are available on _Gadi_ inside the directory `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>`. They are also symlinked in `~/cylc-run/<suite-ID>`.

Author: replace `inside` with `in`

The RNS output data can be found in the directory `/scratch/$PROJECT/$USER/cycl-run/<suite-ID>/share/cycle`, grouped for each [cycle](#change-run-length-and-cycling-frequency).<br>

> Within the `cycle` directory, outputs are divided into multiple nested subdirectories in the format `<nested_region_name>/<science_configuration>/<nest_name>`, with [`<nested_region_name>`](#change-the-nested-region-name) and `<nest_name>` referring to the respective configurable options. The `<science_configuration>` is usually one among `GAL9` or `RAL3.2`, depending on the [nest resolution]({{ model_configurations }}/#model-components).

Author: edit: `The `<science_configuration>` is usually `GAL9` or `RAL3.2`, depending on the [nest resolution]({{ model_configurations }}/#model-components).`

Each `<nest_name>` directory has the following subdirectories:

- `ics` &rarr; initial conditions
- `lbcs` &rarr; lateral boundary conditions
- `um` &rarr; model output data

> Therefore, the model output data for the first cycle (`20220226T0000Z`) of the example in this documentation (`Lismore` `nested_region_name`, using a `RAL3P2` `science_configuration` and `d0198` as a `nest_name`) is stored in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/cycle/20220226T0000Z/Lismore/d0198/RAL3P2/um`.

Author: edit: `For example, the model output data for the first cycle (`20220226T0000Z`) of the Lismore experiment (`Lismore` `nested_region_name`, using a `RAL3P2` `science_configuration` and `d0198` as a `nest_name`) can be found in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/cycle/20220226T0000Z/Lismore/d0198/RAL3P2/um`.`

The RNS output data files are typically in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

!!! tip
    To change which output variables are produced, refer to [Change the output variables](#change-the-output-variables)

### Edit {{ model }} configuration
!!! tip
    Due to the presence of two distinct suites ([RAS](#ras) and [RNS](#rns)), specific modifications might be required in either one or both.<br>

>     To clarify which suite needs editing, each modification paragraph includes a bullet point specifying the relevant suite to edit.

Author: edit: `For each of the following possible modifications, the relevant suite that needs editing is listed as a bullet point.`

This section describes how to modify the {{ model }} configuration.

The modifications discussed in this section can change the way the RAS and RNS are run, or how specific [model components] are configured and coupled together.

In general, ACCESS modelling suites can be edited either by directly modifying the configuration files within the suite directory, or by using the [_Rose_ GUI](#rosegui).

!!! warning

>     Unless you are a very expert user, directly modifying configuration files is usually discouraged to avoid incurring in errors.

Author: edit: ` Unless you are an experienced user, directly modifying configuration files is usually discouraged to avoid encountering errors.`

##### Rose GUI {: #rosegui }
To open the [_Rose_](#rose) GUI, run the following command from within the [suite directory](#suitedir): 
```
rose edit &
```

!!! tip
    The `&` is optional. It allows the terminal prompt to remain active while running the `Rose` GUI as a separate process in the background.

#### Change run length
- **RNS**<br>

>     The RNS runs in multiple [PBS jobs][PBS job] submissions, each one constituting a _cycle_. The job scheduler automatically resubmits the suite every chosen _cycling frequency_, until the total _run length_ is reached.<br>

Author: remove comma `_cycling frequency_ until`

    
    !!! warning
        The _cycling frequency_ is currently set to `24` hours (1 day) and should be left unchanged to avoid errors.<br>
        This also means the model will run for a minimum of 1 day.
      
    The _run length_ is calculated using the `INITIAL_CYCLE_POINT` and `FINAL_CYCLE_POINT` fields.<br>
    Both these fields use [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date format, with `FINAL_CYCLE_POINT` also accepting relative [ISO 8601 Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations).

    !!! warning
        `INITIAL_CYCLE_POINT` and `FINAL_CYCLE_POINT` define all the [_Cylc_ cycle points](https://cylc.github.io/cylc-doc/7.9.3/html/terminology.html?highlight=cycle%20point#cycle-points) that are set within the experiment run.<br>
        The model will always run for a full _cycling frequency_ (1 day) for each _Cylc_ cycle point.<br>
        This means, for example, that with `INITIAL_CYCLE_POINT` set to `20220226T0000Z`, and `FINAL_CYCLE_POINT` set to `+P1D` (plus 1 day), 2 _Cylc_ cycle points will be set (`20220226T0000Z` and `20220227T0000Z`). Therefore, the model will run for a total of 2 days!<br>

>         To avoid running the model for longer that desired, we suggest adding `-PT1S` (minus 1 second) to the relative duration specified in the `FINAL_CYCLE_POINT` (related example below).

Author: edit: ` ...(refer to example below).`

        {: #run-length-mismatch }

>     To modify these parameters, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Nesting Suite &rarr; Cycling options_, edit the related field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.<br>

Author: remove comma `To modify these parameters within...`

>     For example, to run the experiment for 2 days, starting the 5th April 2000, set `INITIAL_CYCLE_POINT` to `20000405T0000Z` and `FINAL_CYCLE_POINT` to `+P2D-PT1S` (due to the [run length mismatch](#run-length-mismatch)).

Author: edit: `... for 2 days starting on the 5th April, 2000, set...`

#### Change the land-surface initial conditions source
- **RNS**<br>
    To change the [land-surface initial conditions source]({{ model_configurations }}/#land-surface-initial-conditions-source), within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Nesting Suite &rarr; Driving model setup_, edit the `NCI_HRES_ECCB` field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

    For example, to get the land-surface initial conditions from the `BARRA-R2` dataset, set the `NCI_HRES_ECCB` field to `BARRA2-R`.

!!! warning

>     When changing the land-surface initial conditions source, it is important to always ensure that the nested region's nest configuration aligns with the [nest configuration requirements](#nest-configuration-requirements).

Author: edit: `When changing the land-surface initial conditions source, it is important to ensure that the configuration of the nested region aligns with the [nest configuration requirements](#nest-configuration-requirements).`

> #### Change the simulation region

Author: I would add a bullet point with `RAS and RNS` to be consistent with other "modification paragraphs"

> In {{ model }}, users can perform simulations for distinct regions of Earth by configuring specific parameters for each domain of interest (referred to as _nested region_).<br>

Author: edit: `In {{ model }}, users can perform simulations for a particular region of the Earth by configuring specific parameters for each domain of interest (referred to as _nested region_).<br>`

In {{ model }}, the following parameters are supported to configure the nested regions:

- [Nested region name](#change-the-nested-region-name)
- [Nested region position](#change-the-nested-region-position)
- [Nested region's nest configuration](#change-the-nested-regions-nest-configuration)

!!! warning
    Domain-specific changes need to be consistent between RAS and RNS. Therefore, for each of the configuration parameters listed above, consistent changes to both RAS and RNS will be required.

##### Change the nested region name {: .no-toc }
- **RAS**<br>

>     To change a nested region name, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Regional Ancillary Suite &rarr; Nested region 1 setup_, edit the `rg01_name` field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

Author: edit: break into 2 sentences: `... Edit the...`

    For example, to set the name of the nested region to `Darwin`, set the `rg01_name` field to `Darwin`.

- **RNS**<br>

>     Changing the RAS nested region name changes the [RAS output path](#ras-output-files). As a consequence, the following changes are required within the RNS:

Author: edit: `Consequently, the following...`

    
    - **Ancillary directory**<br>

>         To change the first nest ancillary directory, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Nesting Suite &rarr; Nested region 1 setup &rarr; Resolution 1 setup_, change the `rg01_rs01_ancil_dir` field by replacing `Lismore` with the chosen RAS nested region name, and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.<br>

Author: edit: break into 2 sentences: `... Change the...`

        The same step needs to be repeated for:
        
        - _suite conf &rarr; Nesting Suite &rarr; Nested region 1 setup &rarr; Resolution 2 setup_ `rg01_rs02_ancil_dir` field
        - _suite conf &rarr; Nesting Suite &rarr; Driving model setup_ `dm_ec_lam_ancil_dir` field

        For example, if the RAS nested region name was set to `Darwin`, replace `Lismore` in the `rg01_rs01_ancil_dir`, `rg01_rs02_ancil_dir` and `dm_ec_lam_ancil_dir` fields with `Darwin`.

    - **RNS nested region name**<br>

>         To change the nested region name, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Nesting Suite &rarr; Nested region 1 setup_, edit the `rg01_name` field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

Author: edit: break into 2 sentences:` ... Edit the `

        For example, to set the name of the nested region to `Darwin`, set the `rg01_name` field to `Darwin`.

        !!! tip
            Changing the RNS nested region name is not strictly necessary, but it affects the [RNS outputs path](#rns-output-files). Therefore, for consistency, it is strongly recommended for RAS and RNS to have the same nested region names.

##### Change the nested region position {: .no-toc }
The nested region position is usually defined by the latitude and longitude coordinates of the nested region centre.

- **RAS**<br>

>     To change the nested region centre, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Regional Ancillary Suite &rarr; Nested region 1 setup_, edit the `rg01_centre` field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.

Author: edit: break into 2 sentences: `... Edit the `

    For example, to set the centre of the nested region to `-12.4` / `130.8`, set the `rg01_centre` field to `-12.4` / `130.8`.

!!! warning

>     When changing the nested region position, it is important to always ensure that the nested region's nest configuration aligns with the [nest configuration requirements](#nest-configuration-requirements).

Author: [pp](edit: When changing the lnested region position, it is important to ensure that the configuration of the nested region aligns with the [nest configuration requirements](#nest-configuration-requirements).)

##### Change the nested region's nest configuration {: .no-toc }
Each nested region can contain multiple [nests]({{model_configurations}}/#nesting) (referred to as _Resolutions_ within the RAS and RNS), each of them being a separate domain where the simulation experiment is carried out.<br>

> Typically, nests within the same nested region are arranged concentrically, with their dimensions and resolutions decreasing towards the innermost nests.

Author: edit: `Typically, nests (resolutions) within the same nested region are arranged concentrically, with increasingly smaller dimensions and higher resolution towards the innermost nests.`

<div markdown id='nest-configuration-requirements'>
!!! warning

>     Currently, {{model}} only supports specific nest configurations, that meet the following criteria:

Author: edit: remove comma `... nest configurations that meet the following criteria:`

>     The grid points of the RAS first inner nest (i.e.: _Resolution 2_, because _Resolution 1_ always corresponds to the outer ERA5 domain) need to match the [land-surface initial conditions dataset]({{model_configurations}}/#land-surface-initial-conditions-source) grid-points. Therefore, configurations (position, dimension and resolution) for the RAS first inner nest (_Resolution 2_) need to be modified accordingly. Note that the position of a nest is also influenced by the [nested region position](#change-the-nested-region-position).

Author: edit: `The grid points of the RAS first inner nest (i.e., _Resolution 2_, because _Resolution 1_ corresponds to the outer ERA5 domain) need to match the ) grid-points of the [land-surface initial conditions dataset]({{model_configurations}}/#land-surface-initial-conditions-source). Therefore, configurations (position, dimension and resolution) for the RAS first inner nest (_Resolution 2_) need to be modified accordingly. Note that the location of the nest is also influenced by the [nested region position](#change-the-nested-region-position).`

Reviewer 2: @heidinett 
This is the outdated version, would you mind reviewing the updated one please?

</div>

#### Change the output variables
[UM](/models/model_components/atmosphere/#unified-model-um) outputs are usually provided as a list of [STASH](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_C04.pdf) variables.<br>

> Manually specifying each STASH variable can be complex. To simplify the selection process for commonly used climate analysis variables, predefined groups of STASH variables, known as _stashpacks_, have been created.

Author: edit: `To simplify the selection process for commonly used climate analysis variables, there are predefined groups of STASH variables known as _stashpacks_.`

- **RNS**<br>

>     To toggle a _stashpack_, within the [Rose GUI](#rosegui) navigate to _suite conf &rarr; Nesting Suite &rarr; Nested region 1 setup &rarr; Resolution 1 setup &rarr; Config 1 setup_, toggle a specific _stashpack_ within the `rg01_rs01_m01_stashpack` field and click the _Save_ button ![Save button](/assets/run_access_cm/save_button.png){: style="height:1em"}.<br>

Author: edit: break into 2 sentences: `... Toggle a specific...`

    Similar steps can be repeated for the _suite conf &rarr; Nesting Suite &rarr; Nested region 1 setup &rarr; Resolution 2 setup &rarr; Config 2 setup_ `rg01_rs02_m02_stashpack` field.<br>

>     For example, to enable stashpack `6` (that includes variables such as wind gust, mean sea level pressure and rainfall amount, for every model timestep) in all nests, set the `6th` button of both `rg01_rs01_m01_stashpack` and `rg01_rs02_m02_stashpack` fields to `true`.

Author: edit: `to enable `stashpack 6`

{% set model = "ACCESS-rAM3" %}
{% set ras_id = "u-dg767" %}
{% set rns_id = "u-dg768" %}
{% set mosrs_config_ras = "https://code.metoffice.gov.uk/trac/roses-u/browser/d/g/7/6/7/trunk" %}
{% set mosrs_config_rns = "https://code.metoffice.gov.uk/trac/roses-u/browser/d/g/7/6/8/trunk" %}
{% set release_notes = "https://forum.access-hive.org.au/t/access-ram3-release-information/4308" %}
[PBS job]: https://opus.nci.org.au/display/Help/4.+PBS+Jobs
[model components]: /models/configurations/access-ram/#model-components
[model configurations]: /models/configurations/access-ram
[gadi]: https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview

[:notepad_spiral:{: class="twemoji icon-before-text"} {{ model }} release notes]({{release_notes}}){: class="text-card"}
[![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} RAS configuration]({{mosrs_config_ras}}){: class="text-card" style=""}

> [![Met Office](/assets/met_office_logo.png){: class="icon-before-text  white-background"} RNS configuration]({{mosrs_config_rns}}){: class="text-card"}

Reviewer 3: Is there a way to make these smaller? And also potentially put them on the same line? On a laptop screen they really dominate.

<img width="1492" alt="Screenshot 2025-03-18 at 12 29 51 pm" src="https://github.com/user-attachments/assets/59d0638c-7a99-4c13-b007-38e941d14ad2" />

Arguably it should also be possible to change to just "Release Notes" as the model is implied.

Reviewer 2: Yes, I think all on the same line might not be idea (if they are 3 or more), but their size can definitely be reduced and their layout set so that they occupy one line if they are `<=2` and more line if they are `>2`.

Also changing it to "Release notes" makes sense.
I will also add the same "Release notes" card for the other released models (ACCESS-OM and ACCESS-ESM).

I will address this.

Reviewer 2: Addressed in [5f944801f852fdb54f91408d15a17c158eaf5268](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/5f944801f852fdb54f91408d15a17c158eaf5268)

> # Run {{ model }}

Reviewer 3: Not sure where it is injected, but the info panel for the code block stuff is a bit intrusive right at the top of the docs IMO.

Could we move it down to the part where the code blocks are first used? I guess the Quick Start Guide?

Reviewer 2: I agree it's a bit intrusive. 
Not sure if placing it where the code bocks are first used is a good solution though. 

How about making it a tooltip that pops up only when the little terminal icon at the right of the page title (the one you can click to toggle animations) is hovered over?

Reviewer 2: Fixed in #906.

    ```
    echo "<name>.${USER}.<project>.ps.gadi.nci.org.au" > ~/.persistent-sessions/cylc-session
    ```
    substituting `<name>` with the name given to your _persistent session_, and `<project>` with the project assigned to it.

    !!! tip
        This step should only be done once

    For further instructions on assigning the target _persistent session_, refer to the [detailed guide](#specify-target-persistent-session).

- **Rose/Cylc setup**<br>
    To get the required _Rose/Cylc_ setup, run:
    ```
    module use /g/data/hr22/modulefiles
    module load cylc7
    ```

    For further instructions on getting the _Rose/Cylc_ setup, refer to the [detailed guide](#rosecylcmosrs-setup).

- **MOSRS authentication**<br>
    Authenticate using your MOSRS credentials:
    ```
    mosrs-auth
    ```

    For further instructions on MOSRS authentication, refer to the [detailed guide](#mosrs-authentication).

### Regional Ancillary Suite (RAS) {: .no-toc }
1. **Copy the RAS from UKMO**<br>
    ```
    rosie checkout {{ ras_id }}
    ```

    For further instructions on getting the RAS configuration, refer to the [detailed guide](#get-the-ras-configuration).

2. **Run the RAS**<br>

>     From within the RAS directory:

Reviewer 3: I think it would be better to be **very explicit** which is possible in this case as we're talking about a specific suite, e.g.
```
cd ~/cylc-run/u-dg767
suite run
```

Reviewer 2: Changed to 
```
cd ~/roses/u-dg767
rose suite-run
```
in [fc78870c53f57ac78d91d816844fb3daa06ea1ed](https://github.com/ACCESS-NRI/access-hive.org.au/pull/857/commits/fc78870c53f57ac78d91d816844fb3daa06ea1ed)

[model configurations]: /models/configurations/access-ram

!!! info
    {{ model }} is a place-holder name until the final release name is chosen.

> [:fontawesome-brands-github:{: class="twemoji icon-before-text"} {{ model }} configuration]({{github_configs}}){: class="text-card"}

Reviewer 4: I would move this config card above the info note above so that the info note appears just above the page title. 
And the note about the code blocks that you mentioned in todays meeting automatically comes up on the page because there are terminal animations on the page, probably best to leave it there.

Author: Thanks @flicj191! I moved the config card as you suggested. 
Re: note about code block, it would be great if I could place it in the Detailed guide (at the moment it's in the Quick Start where there are no terminal animations).

Reviewer 4: Looks like its automatically set to appear under the page heading, that's why I've added a page heading and changed others to subheadings in the suggestions above. I guess a hack right now would be to make detailed guide the page heading, then all the headings before that wouldn't appear in the page navigation

Reviewer 4: You can commit the ones you want from above from the files tab in a batch. moving the config card without the added page title wouldn't put the name note above the page title

> 

Reviewer 4: ```suggestion
# Run {{ model }}
```

{{ model }} is an ACCESS-NRI-supported configuration of the UKMO Regional Nesting Suite. Driven by ERA5, the configuration comprises a 2-level nest; the outer level uses GAL9 with a resolution of 10km or 11km (depending on the choice of land-surface initial conditions), and the inner nest that focuses on the region of interest uses RAL3.2 with 2.2 km resolution. 

{{ model }} comprises a Regional Ancillary Suite (RAS) `{{ ras_id }}` to generate ancillary files needed to run the Regional Nesting Suite (RNS) `{{ rns_id }}` for the domain of interest.

<!--
TODO
{: style="color:red"}
<!-- Can we put a configuration here taken from MOSRS instead of GitHub? SVN? -->

The following *Quick Start* guide is aimed at experienced users wanting to run {{ model }}. If you would prefer more detailed instructions, please refer to the *Detailed Guide* below.

<!-- Quick start guide for experienced users -->

> # Quick Start

Reviewer 4: ```suggestion
## Quick Start
```

- **Join relevant NCI projects:**<br> _access_, _hr22_, _ki32_, _ki32\_mosrs_, _rt52_, _zz93_ and _vk83_ 

- **Start a new persistent session on *Gadi***<br> using a login node or an ARE terminal instance:
```
persistent-sessions start <name>
```
This will use your default project. Use option `-p` to assign a different project to the persistent session:
```
persistent-sessions start -p <project> <name>
```
- **Cylc setup**<br>
To get the required _Cylc_ setup, run:
```
module use /g/data/hr22/modulefiles
module load cylc7
```
- **MOSRS authentication**<br>
Authenticate using your MOSRS credentials:
```
mosrs-auth
```
### Regional Ancillary Suite (RAS)
1. **Copy the RAS from UKMO**<br>
    Local-only copy: 
```
rosie checkout {{ ras_id }}
```
    New copy both _locally_ and _remotely_: 
```
rosie copy {{ ras_id }}
```
2. **Run the RAS**<br>
From within the suite directory:
```
rose suite-run
```

### Regional Nesting Suite (RNS)
1. **Copy the RNS from UKMO**<br>
    Local-only copy: 
```
rosie checkout {{ rns_id }}
```
    New copy both _locally_ and _remotely_: 
```
rosie copy {{ rns_id }}
```
2. **Run the RNS**<br>
From within the suite directory:
```
rose suite-run
```
____

> ----
> * * * 

Reviewer 4: ```suggestion
!!! tip
    You've completed the quick reference steps! For more comprehensive instructions keep reading below.
```
Could use something like this to break up the sections on the page? feel free to delete

> # Detailed Guide

Reviewer 4: ```suggestion
## Detailed Guide
```
